### PR TITLE
Added Anisotropies in HNL decay to 2 identical flavour leptons and a …

### PR DESCRIPTION
**Pull request by LuisPelegrina**
…neutrino. Pull request dependent on sbndcode Pull request #324 (https://github.com/SBNSoftware/sbndcode/pull/324)

### CODE REVIEW

File: sbncode/EventGenerator/MeVPrtl/Tools/HNL/AnisotropicThreeBodyDecay.cpp [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+  
+  for (int i=0;i<6;i++)
+    {
+      CD[i]=0;
+    }
+
+  double gl;
+  double gr=GR();
+  //Calculate the contribution of each mixing matrix element and sum them 
+  for (int i=0;i<3;i++)
```
Comment (jicrespo) [inline sbncode/EventGenerator/MeVPrtl/Tools/HNL/AnisotropicThreeBodyDecay.cpp:73]: Table 3 from the reference only applies to the case of nu_mu - N mixing. This for loop over all matrix elements is not correct.

**Review by jicrespo — CHANGES_REQUESTED** [review]
I have not finished the complete review of the code but I have already identified a few issues that should be addressed, along format changes. I will finish the review once these issues are addressed.

File: sbncode/EventGenerator/MeVPrtl/Tools/HNL/AnisotropicThreeBodyDecay.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+//Implementation of HNL Three Body Decays Anisotropies
+//Valid as long as the HNL is decaying into a neutrino and identical final-state charged leptons
+//See arXiv:2104.05719 for more details @LuisPelegrina
+
+
+namespace evgen {
+namespace ldm {
+namespace AnThreeBD {
```
[the remark this answers is not in the document]
Comment (LuisPelegrina) [reply sbncode/EventGenerator/MeVPrtl/Tools/HNL/AnisotropicThreeBodyDecay.h:14]: Just for organizing myself

File: sbncode/EventGenerator/MeVPrtl/config/HNL/hnl.fcl [hunk 3]
```diff
@@ -33,6 +33,7 @@ decay_hnl: {
   tool_type: HNLMakeDecay
   Decays: ["mu_pi"]
   Majorana: true
+  DecayIsThreeBodyAnisotropic: true
```
[the remark this answers is not in the document]
Comment (LuisPelegrina) [reply sbncode/EventGenerator/MeVPrtl/config/HNL/hnl.fcl:36]: This Option does not affect the mupi decay, as it is implemented in a different function than HNLMakeDecay::DecayFinalState HNLMakeDecay::NuDiLep

File: sbncode/EventGenerator/MeVPrtl/config/HNL/hnl.fcl [hunk 4]
```diff
@@ -47,6 +48,7 @@ decay_tau_hnl: {
   Decays: ["nu_mu_mu"]
   WidthDecays: ["nu_nu_nu", "nu_mu_mu", "nu_e_e", "nu_pi0", "nu_eta", "nu_etap", "nu_rho0"]
   Majorana: false
+  DecayIsThreeBodyAnisotropic: true
```
[the remark this answers is not in the document]
Comment (LuisPelegrina) [reply sbncode/EventGenerator/MeVPrtl/config/HNL/hnl.fcl:51]: Tau decays are not implemented in the Simulator as the required HNL mass for a nutautau final state is not achievable by meson decay, it is not supported even for a isotropic decay

File: sbncode/Supera
Comment (jicrespo) [inline sbncode/Supera:1]: Is this part of the commit?

**Review by jicrespo — APPROVED** [review]
The sbncode/Supera commit seems not related to this pull request. @LuisPelegrina please check before merging.

**Comment by rodralva** [comment]
trigger build https://github.com/SBNSoftware/sbndcode/pull/324

**Comment by jicrespo** [comment]
trigger build https://github.com/SBNSoftware/sbndcode/pull/324

**Pull request merged**